src/components/navigation/MenuButton.tsx

```diff
@@ -0,0 +1,49 @@
+'use client';
+
+type TMenuButton = {
```

Comment:
Coming with a little explanation here! Since the project currently has interfaces and types definitions, where interfaces are prefixed with `I` such as `IButton`, it's a common pattern that the types are prefixed by `T` char - that's why I used it here. 

Long term, I would suggest to keep using types instead of Interfaces, because of the fact that the rules of merging them are a bit risky (e.g when the project has multiple interfaces with the same name, or when applying unions & intersections, where it's not possible to use interfaces). There are a lot of major and minor differences, but (TLDR;) types are just better for smaller projects because of complexity of interfaces.